### PR TITLE
Fix: unstable sort skips/duplicates issues across pages

### PR DESCRIPTION
**Pull request by DuckDuckWhale**
When viewing issues in sorted order, some issues are duplicated across pages and some are missing. This is caused by the lack of tie-breakers in database queries, making pagination inconsistent.

Backport of #18094.

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

File: models/issue.go [hunk 1]
```diff
@@ -1145,35 +1145,39 @@ type IssuesOptions struct {
...
+		sess.Desc("issue.updated_unix").Desc("issue.id")
 	case "leastupdate":
-		sess.Asc("issue.updated_unix")
+		sess.Asc("issue.updated_unix").Asc("issue.id")
 	case "mostcomment":
-		sess.Desc("issue.num_comments")
+		sess.Desc("issue.num_comments").Desc("issue.id")
 	case "leastcomment":
-		sess.Asc("issue.num_comments")
+		sess.Asc("issue.num_comments").Desc("issue.id")
```
Comment (KN4CK3R) [inline models/issue.go]:
```suggestion
		sess.Asc("issue.num_comments").Asc("issue.id")
```

File: models/issue.go [hunk 2]
```diff
@@ -1145,35 +1145,39 @@ type IssuesOptions struct {
...
 	case "nearduedate":
 		// 253370764800 is 01/01/9999 @ 12:00am (UTC)
 		sess.Join("LEFT", "milestone", "issue.milestone_id = milestone.id").
 			OrderBy("CASE " +
 				"WHEN issue.deadline_unix = 0 AND (milestone.deadline_unix = 0 OR milestone.deadline_unix IS NULL) THEN 253370764800 " +
 				"WHEN milestone.deadline_unix = 0 OR milestone.deadline_unix IS NULL THEN issue.deadline_unix " +
 				"WHEN milestone.deadline_unix < issue.deadline_unix OR issue.deadline_unix = 0 THEN milestone.deadline_unix " +
-				"ELSE issue.deadline_unix END ASC")
+				"ELSE issue.deadline_unix END ASC").
+			Desc("issue.id")
```
Comment (KN4CK3R) [inline models/issue.go:1168]:
```suggestion
			Asc("issue.id")
```

File: models/issue.go (hunk 1 above)
Comment (DuckDuckWhale) [reply models/issue.go]: The rationale here is that when sorting by the number of comments the order of creation do not matter, so in both cases fallback to default, i.e. descending order, i.e. newest first.

File: models/issue.go (hunk 2 above)
Comment (DuckDuckWhale) [reply models/issue.go:1168]: The rationale here is that when sorting by deadline the order of creation do not matter, so in both cases fallback to default, i.e. descending order, i.e. newest first.

Comment (DuckDuckWhale) [reply models/issue.go:1168]: Addressed in original PR.

File: models/issue.go (hunk 1 above)
Comment (DuckDuckWhale) [reply models/issue.go]: Addressed in original PR.

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**